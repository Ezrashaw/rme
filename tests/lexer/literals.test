123
123.
.123
12.3
12..3
123.0.1
-1
-.1
999999999999999999999999999999999999999

true
false
2true
true2
2false
false2
True
False
let_true
true_let