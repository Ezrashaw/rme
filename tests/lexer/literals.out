Literal(Float(123.0))@0..3
Literal(Float(123.0))@4..8
Literal(Float(0.123))@9..13
Literal(Float(12.3))@14..18
[31;1;3merror[0m: invalid float literal
[38;5;244;1m  |[0m 
[38;5;244;1m5 |[0m 12..3
[38;5;244;1m  |[0m [96;1m^^^^^[0m
[31;1;3merror[0m: invalid float literal
[38;5;244;1m  |[0m 
[38;5;244;1m6 |[0m 123.0.1
[38;5;244;1m  |[0m [96;1m^^^^^^^[0m
Minus@33..34
Literal(Float(1.0))@34..35
Minus@36..37
Literal(Float(0.1))@37..39
Literal(Float(inf))@40..79
Literal(Bool(true))@81..85
Literal(Bool(false))@86..91
Literal(Float(2.0))@92..93
Literal(Bool(true))@93..97
Identifier("true2")@98..103
Literal(Float(2.0))@104..105
Literal(Bool(false))@105..110
Identifier("false2")@111..117
Identifier("True")@118..122
Identifier("False")@123..128
Identifier("let_true")@129..137
Identifier("true_let")@138..146
